src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs

```diff
@@ -196,10 +196,16 @@ private static DependencyGraphSpec ToDependencyGraphSpec(ProjectNames projectNam
                     .Any(tfm => tfm.Version >= Version20);
                 var toolFramework = isNetCore20 ? CommonFrameworks.NetCoreApp20 : CommonFrameworks.NetCoreApp10;
 
+                var packagesPath = GetRestoreProjectPath(projectRestoreInfo.TargetFrameworks);
+                var fallbackFolders = GetRestoreFallbackFolders(projectRestoreInfo.TargetFrameworks).ToList();
```

Comment:
calling ToList() causes the underlying collection to be iterated , sometimes unnecessarily. 
CHange GetRestoreFallbackFolders to return IList instead of string[] and then just pass this as is. Apply the same at other places too.